mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -1932,6 +1932,20 @@ OpFoldResult LLVM::ExtractValueOp::fold(FoldAdaptor adaptor) {
     getContainerMutable().set(extractValueOp.getContainer());
     return getResult();
   }
+
+  {
+    DenseElementsAttr constval;
+    matchPattern(getContainer(), m_Constant(&constval));
+    if (constval && constval.getElementType() == getType()) {
+      if (isa<SplatElementsAttr>(constval)) {
```

Comment:
Could the splat have multi dim shape such as <2x2xi32> and the getPosition() having one element which indexes only in the first dimension?